### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #10, #43

> :warning: Depends on PR #36.

Docs:

- [x] `Config/ClassConfig`
- [x] `Config/MethodConfig`
- [x] `Config/ParameterConfig`
- [x] `Config/TypeConfig`
- [x] `ClassCache`
- [x] `ClassConfigFactory`
- [x] `ClassRenderer`
- [x] `ObjectProxy`
- [x] `ProxyManager`
- [x] `ProxyTrait`
- [x] README

### CODE REVIEW

File: src/ClassCache.php [hunk 1]
```diff
@@ -8,55 +8,101 @@
...
-        }
-        file_put_contents($this->getClassPath($className, $classParent), "<?php\n\n" . $classDeclaration, LOCK_EX);
+        file_put_contents($this->getClassPath($className, $baseProxyClassName), "<?php\n\n" . $classContents, LOCK_EX);
     }
 
-    public function get(string $className, string $classParent): ?string
+    /**
+     * Reads proxy class contents.
+     *
+     * @param string $className {@see set()}
```
[the remark this answers is not in the document]
Comment (arogachev) [reply src/ClassCache.php:45]: :ok_hand: :heavy_check_mark:

File: src/ClassCache.php [hunk 2]
```diff
@@ -8,55 +8,101 @@
...
+     * @param string $baseProxyClassName {@see set())
```
[the remark this answers is not in the document]
Comment (arogachev) [reply src/ClassCache.php:46]: :ok_hand: :heavy_check_mark:

File: src/ClassCache.php [hunk 3]
```diff
@@ -8,55 +8,101 @@
...
         return $content;
     }
 
-    public function getClassPath(string $className, string $classParent): string
+    /**
+     * Gets full path to a class. For example: `/tmp/Yiisoft/Tests/Stub/GraphInterface.MyProxy.php` or
+     * `/tmp/Yiisoft/Tests/Stub/Graph.MyProxy.php`. Additionally, checks and prepares (if needed) {@see $cachePath} for
+     * usage (@see FileHelper::ensureDirectory()}.
+     *
+     * @param string $className {@see set()}
```
[the remark this answers is not in the document]
Comment (arogachev) [reply src/ClassCache.php:69]: :ok_hand: :heavy_check_mark:

File: README.md [hunk 4]
```diff
@@ -1,19 +1,186 @@
...
+
+$path = sys_get_temp_dir();
+$manager = new ProxyManager($path);
+/** @var Car|ObjectProxy $object */
+$object = $manager->createObjectProxy(
+    Car::class, // Pass class instead of interface here. 
+    ObjectProxy::class, 
+    [new Car()]
+);
+$object->horsepower(); // Outputs "1".
```
[the remark this answers is not in the document]
Comment (arogachev) [reply README.md:90]: Thanks for noticing this. Agree. :ok_hand:  Fixed. :heavy_check_mark:

**Review by samdark — APPROVED** [review]

**Pull request merged**